glooshot/docs/tutorial/bookinfo_tutorial.md

```diff
@@ -0,0 +1,265 @@
+  - Now when you refresh the page, the stars should always be red.
+```bash
+supergloo apply routingrule trafficshifting \
+    --namespace glooshot \
+    --name reviews-v4 \
+    --dest-upstreams glooshot.default-reviews-9080 \
+    --target-mesh glooshot.istio-istio-system \
+    --destination glooshot.default-reviews-v4-9080:1
+```
+
+
+### Create an experiment
+
+- Create a simple experiment with `kubectl`
+  - We will create a fault on the ratings service such that it always returns `500` as a response code.
+  - We will run this experiment with the following conditions:
+    - The prometheus query `scalar(sum(istio_requests_total{ source_app="productpage",response_code="500"}))` must not exceed a threshold of 10.
+    - The experiment should expire after 600 seconds
+  - Execute the command below to create this experiment
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: glooshot.solo.io/v1
+kind: Experiment
+metadata:
+  name: abort-ratings-metric
+  namespace: default
+spec:
+  spec:
+    duration: 600s
+    failureConditions:
+      - prometheusTrigger:
+          customQuery: |
+            scalar(sum(istio_requests_total{ source_app="productpage",response_code="500"}))
+          thresholdValue: 10
+          comparisonOperator: ">"
+    faults:
+    - destinationServices:
+      - name: default-ratings-9080
+        namespace: glooshot
+      fault:
+        abort:
+          httpStatus: 500
+        percentage: 100
+    targetMesh:
+      name: istio-istio-system
+      namespace: glooshot
+EOF
+```
+
+- Refresh the page, you should now see a failure: none of the reviews data is rendered
+- Refresh the page about 10 more times.
+- Within 15 seconds after the threshold value is exceeded you should see the error go away. The experiment stop condition has been met and the fault that caused this cascading failure has been removed.
+- Inspect the experiment results with the following command:
+```bash
+kubectl get exp abort-ratings-metric -o yaml
+```
+
+- You should see something like this:
+```bash
+  result:
+    failureReport:
+      comparison_operator: '>'
+      failure_type: value_exceeded_threshold
+      threshold: "10"
+      value: "20"
+    state: Failed
+    timeFinished: "2019-05-13T17:27:49.799279861Z"
+    timeStarted: "2019-05-13T17:27:34.650136785Z"
+```
+- Note that the state reports the experiment has "Failed". This is because the experiment was terminated because a threshold value was exceeded. If the experiment had been terminiated by a timeout, it would be in state "Succeeded".
+  - Experiments that fail, such as this one, indicate that our service is not as robust as we would like.
+- The experiment also reports the exact value that was observed, which caused the failure. Note that the value is 20, which exceeds our limit of 10. The reason for this is that Prometheus gathers metrics every 15 seconds. The metric value may rise above the limit in the time it takes for Prometheus to report the exceeded limit.
```

Comment:
ha! okay cool.. just seeing this explanation now... up to you to re-order or keep it.. glad you do explain it here though